### DISCUSSION TITLE
Waiting for server

### DISCUSSION THREAD

**Discussion by marea69** [inventree/InvenTree#2607] [General]
Hello
I would like to optimize this function if possible. When I click on the button, create new stock item, the time between the click and the opening of the form is about 6 seconds. (Waiting for server)
No problem if I create a new part instead. You understand that if I have to insert more than 200 changes on the stocks the total waiting time is very high

How can I fix this ?

![Cattura](https://user-images.githubusercontent.com/61009409/152750767-10ea542a-6ad9-4710-b292-de0d06d1b3a9.PNG)

**Comment by SchrodingersGat** [comment]
@marea69 I see from your screenshot that you are running version `0.5.x`. The [imminent release](https://inventree.readthedocs.io/en/latest/releases/0.6.0/) of `0.6.0` provides a significant refactor of all user-facing forms. In particular, the StockItem creation forms have been overhauled for a huge speed improvement. You can check out the [particular changes here](https://github.com/inventree/InvenTree/pull/2198)

The [demo server](https://demo.inventree.org/) is running the latest code, you can check it out there to see how the new forms work.
